src/utils/plotSpecs/generateViolinSpec.js

```diff
@@ -477,19 +477,20 @@ const generateData = (
 
   cellSetsIds.forEach((cellSetId) => {
     const currentCellIds = Array.from(properties[cellSetId].cellIds);
-
     currentCellIds
       .filter(shouldBeDisplayed)
       .forEach((cellId) => {
-        const cell = {
-          group: cellSetId,
-          y: selectedExpression[cellId],
-        };
+        if (selectedExpression[cellId] || selectedExpression[cellId] === 0) {
```

Comment:
could you add a small comment here about why you are checking this? if cellIds included all the unfiltered cells and it shouldn't be the case, shouldn't the fix be to actual check why unfiltered cells make it here?

Reply:
The fact that the unfiltered cells are saved in redux caused another bug before. I'm not sure why it's done (could be on purpose). But I'll add a comment

Reply:
my question is whether we should check why there are unfiltered cells in the redux store instead of just adding this hotfix